examples/C/demo2d.c

```diff
@@ -0,0 +1,62 @@
+/* simple 2d demo of baobzi. to compile & run:
+
+# here we assume baobzi installed in $HOME/local
+export LIBRARY_PATH=$LIBRARY_PATH:$HOME/local/lib
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$HOME/local/lib
+export C_INCLUDE_PATH=$C_INCLUDE_PATH:$HOME/local/include
+
+gcc demo2d.c -o demo2d -lbaobzi -lm && time OMP_NUM_THREADS=1 ./demo2d
+
+(here math libs needed for our user function)
+
+ */
+#include <baobzi.h>
+#include <stdio.h>
+#include <math.h>
+
+
+double f(const double *x, const void *data)
+/* the user func to approximate: must have arguments (x,data)
+   
+   x = 2-element ordinate array, ie x- then y-coords
+   data = user-available parameters array (here unused)
+*/
+{
+  //  return x[0] * x[1];
+  return sin(3.0 + 4*x[0] + 7*x[1]);   // plane wave
+}
+
+
+int main(int argc, char *argv[]) {
+  
+  baobzi_input_t input = {
+    .func = f,                 // ptr to func
+    .data = NULL,
+    .dim = 2,
+    .order = 16,                // high order. (should auto-choose given tol?)
```

Comment:
For speed's sake, it's usually desirable to still use a lower order even when looking for high accuracy (to a point). The tree search is reasonably cheap compared to the Order^dim cost of evaluation. Not sure if auto-selection is reasonable given that.